pkg/cluster/nodes/nodes.go

```diff
@@ -99,3 +100,35 @@ func list(visit func(string, *Node), filters ...string) error {
 	}
 	return nil
 }
+
+// WaitForReady uses kubectl inside the "node" container to check if the
+// master nodes are "Ready".
+func WaitForReady(node *Node, until time.Time) bool {
+	return tryUntil(until, func() bool {
```

Comment:
`tryUntil` might need an extension to treat `0` as an infinite attempts.
to my understanding with the default value of `0` it will not execute the first kubectl call.

would it make sense to add a default value of say ~4 minutes like in kubeadm?

Reply:
I think we should actually default to 0, with no extra waiting.

Why?

Well waiting is mostly a mis-optimization to reduce flakiness of certain tests suites (_glares at k/k e2e_) that do not handle waiting for cluster readiness very well, there's still reasons the code doing tests should handle this. 

Most CI jobs will probably do something like (ideally - we should document this):

- bring up cluster
- do some other tasks (build thing to test etc)
- start testing
- dump logs
- bring down cluster

For that case, it's better to just exit as soon as possible and get on to the other work while the cluster finishes minor readying (CNI really) in the background.


More importantly, the CLI will hang for local users for up to 4 minutes, which is not great. They again will likely do:
- create cluster
- start building something
- make cup of coffee
- run some tests
- ...

We shouldn't hang any more than kubeadm will.
If for some reason this turns out to be problematic, we can add a small default wait and increase it as needed in future versions. 

I'd rather we avoid anything to slow down this part until we actually need it though. So far it's just a single k8s e2e job that has issues with this.

Reply:
> Well waiting is mostly a mis-optimization to reduce flakiness of certain tests suites (glares at k/k e2e) that do not handle waiting for cluster readiness very well, there's still reasons the code doing tests should handle this.

true, random waiting is super nondeterministic and asking for trouble.

so default of 0 means no kubectl calls, as far as i can see it.
sounds good, and hopefully it works. :)